astack.py

```diff
@@ -90,6 +90,9 @@ def find_jstack():
         return default.strip()
     if os.environ.get('JAVA_HOME'):
         return os.path.join(os.environ['JAVA_HOME'], 'bin', 'jstack')
+    jvm_bin_path = os.path.join(Popen('dirname $(readlink /proc/{pid}/exe)'.format(pid=pid), stdout=PIPE, stdin=DEVNULL, shell=True).communicate()[0])
```

Comment:
`dirname` and `readlink` are also implemented in Python (`os.path` and `os` respectively) -- it'd be preferable to use those instead.